File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int a = 0 , b = 0;
        string ans = "";

        while(a != word1.length() && b != word2.length()){
            ans += word1[a];
            ans += word2[b];
            a++;
            b++;
        }
        if(a < word1.length()){
            while(a != word1.length()){
                ans += word1[a];
                a++;
            }
        }
        if(b < word2.length()){
            while(b != word2.length()){
                ans += word2[b];
                b++;
            }
        }
        return ans;
        
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
public 
    String mergeAlternately(String word1, String word2) {
        int a = 0 , b = 0;
        StringBuilder ans = new StringBuilder();

        while(a != word1.length() && b != word2.length()){
            ans.append(word1.charAt(a++));
            ans.append(word2.charAt(b++));
        }

            while(a < word1.length()){
                ans.append(word1.charAt(a++));
            }
    

            while(b < word2.length()){
                ans.append(word2.charAt(b++));
            }
    
        return ans.toString();
        
    }
};